### PR TITLE
patch and test with empty repository

### PR DESCRIPTION
**Pull request by denis-yuen**
**Description**
Add test for completely empty repository refresh and then patch it.

**Review Instructions**
Check that new test is added to test insights on circle ci

**Issue**
https://ucsc-cgl.atlassian.net/browse/SEAB-6717

**Security and Privacy**

None

- [x] Security and Privacy assessed

e.g. Does this change...
* Any user data we collect, or data location?
* Access control, authentication or authorization?
* Encryption features?

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that you pass the basic style checks and unit tests by running `mvn clean install`
- [x] Ensure that the PR targets the correct branch. Check the milestone or fix version of the ticket.
- [x] Follow the existing JPA patterns for queries, using named parameters, to avoid SQL injection
- [x] If you are changing dependencies, check the Snyk status check or the dashboard to ensure you are not introducing new high/critical vulnerabilities
- [x] Assume that inputs to the API can be malicious, and sanitize and/or check for Denial of Service type values, e.g., massive sizes
- [x] Do not serve user-uploaded binary images through the Dockstore API
- [x] Ensure that endpoints that only allow privileged access enforce that with the `@RolesAllowed` annotation
- [x] Do not create cookies, although this may change in the future
- [x] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead.

### CODE REVIEW

File: dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java [hunk 1]
```diff
@@ -1157,6 +1157,12 @@ private GHRef[] getBranchesAndTags(GHRepository repo) throws IOException {
             getBranchesSucceeded = true;
         } catch (GHFileNotFoundException ex) {
             LOG.debug("No branches found for " + repo.getName(), ex);
+        } catch (org.kohsuke.github.HttpException ex) {
+            if (ex.getResponseCode() == HttpStatus.SC_CONFLICT) {
+                // it seems like this is what is returned if there are absolutely no branches or tags
```
Comment (svonworl) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java:1162]: Did some experiments and confirmed that a 409 (SC_CONFLICT) is returned when the repo is empty.  So weird, seems like the intuitive response would be success with an empty result, as is the case when the repo is not empty but no refs are matched.

**Review by svonworl — APPROVED** [review]
